tgui/packages/tgui/interfaces/Newscaster.js

```diff
@@ -0,0 +1,632 @@
+  return (
+    <Section fill title={channelName}>
+      <Stack fill vertical>
+        <Stack.Item grow>
+          {channelCensored ? (
+            <Section>
+              <BlockQuote color="Red">
+                <b>ATTENTION:</b> {CENSOR_MESSAGE}
+              </BlockQuote>
+            </Section>)
+            : (
+              <Section fill scrollable>
+                <BlockQuote italic fontSize={1.2} wrap>
+                  {deconvertHtmlString(channelDesc)}
+                </BlockQuote>
+              </Section>
+            )}
+        </Stack.Item>
+        <Stack.Item>
+          <Box>
+            <Button
+              icon="print"
+              content="Submit Story"
+              disabled={(channelLocked && (channelAuthor !== user.name))
+            || channelCensored}
+              onClick={() => act('createStory', { current: viewing_channel })}
+              mt={1} />
+            <Button
+              icon="camera"
+              selected={photo_data}
+              content="Select Photo"
+              disabled={(channelLocked && (channelAuthor !== user.name))
+            || channelCensored}
+              onClick={() => act('togglePhoto')} />
+            {!!security_mode && (
+              <Button
+                icon="ban"
+                content={"D-Notice"}
+                tooltip="Censor the whole channel and it's \
+                  contents as dangerous to the station. Cannot be undone."
+                disabled={!security_mode || !viewing_channel}
+                onClick={() => act('channelDNotice', {
+                  secure: security_mode,
+                  channel: viewing_channel,
+                })} />
+            )}
+          </Box>
+          <Box>
+            <Button
+              icon="newspaper"
+              content="Print Newspaper"
+              disabled={paper <= 0}
+              onClick={() => act('printNewspaper')} />
+          </Box>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+/** Channel select is the left-hand menu where all the channels are listed. */
+const NewscasterChannelSelector = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [wantedScreen, setWantedScreen] = useLocalState(context, 'wantedScreen', false);
+  const [channelScreen, setChannelScreen] = useLocalState(context, 'channelScreen', false);
+  const {
+    channels = [],
+    viewing_channel,
+    security_mode,
+    wanted = [],
+  } = data;
+  return (
+    <Section
+      minHeight="100%"
+      width={(window.innerWidth - 410) + "px"}>
+      <Tabs vertical>
+        {wanted.map(activeWanted => (
+          <Tabs.Tab
+            pt={0.75}
+            pb={0.75}
+            mr={1}
+            key={activeWanted.index}
+            icon={activeWanted.active ? "skull-crossbones" : null}
+            textColor={activeWanted.active ? "red" : "grey"}
+            onClick={() => { act('toggleWanted',
+              setWantedScreen(true)); }}>
```

Comment:
Horrible code style.

Also if you're toggling wanted via `act`, why are you also using local state? If it's on DM side, then just expose it via ui_data. When having two different sources of truth, you're risking to have a desync between these two states.

```suggestion
            onClick={() => {
              act('toggleWanted');
              setWantedScreen(true);
            }}>
```